chain/jsonrpc/src/lib.rs

```diff
@@ -243,11 +244,32 @@ impl JsonRpcHandler {
         let timer = Instant::now();
 
         let request_method = request.method.clone();
+        let params: RpcQueryRequest = RpcRequest::parse(request.params.clone())?;
```

Comment:
It's not the best approach to parse the params on every request (it's additional parse action).

Could you consider an alternative to use `request.params: serde_json::Value` directly since you have all the data you need there:

something like:

```rust
let request_method = request.params["request_type"];
```